katana/utils/user_utils.py

```diff
@@ -3,6 +3,82 @@
 from django.conf import settings
 from utils.navigator_util import Navigator
 
+from .directory_traversal_utils import join_path, file_or_dir_exists
+
+DOTDATA = '.data'
+WAPPLOGS = 'wapp_logs'
+WAPPSDATA = 'wapps_data'
+USERDATA = 'user_data'
+
+
+class UserData:
+    # constructor for class User_data
+    def __init__(self, wapp_name, username=None):
+        self.username = username
+        self.wapp_name = wapp_name
+
+    def get_user_path(self, request):
+        username = user_authenticated(request)
+        if username is not None:
+            self.username = username
+            top_dir = get_user_home_dir(username)
+            if top_dir is not "":
+                user_dir = join_path(top_dir, USERDATA, username)
+                return user_dir
+            else:
+                print('fetching of top level directory resulted in a empty string')
+                return None
+        else:
+            print('user not authenticated')
+            return None
```

Comment:
Please consolidate all ```return``` into one at the end of the function. Please improve the print statements. We usually use the convention ```print("-- An Error Occurred -- {0}".format(message))```